### PR TITLE
(FM-8217) Port module to litmus

### PR DESCRIPTION
**Pull request by michaeltlombardi**
This commit updates the acceptance testing of the module
to use litmus instead of beaker.

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
I believe this will fail until puppetlabs/puppet_litmus#147 is merged.

File: distelli-manifest.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+team-modules/puppetlabs-package:
```
Comment (tphoney) [inline distelli-manifest.yml]: need to change the title

**Comment by tphoney** [comment]
the tests seem quick, for travis we could combine the el 6+7 test groups

**Comment by michaeltlombardi** [comment]
Acceptance tests are passing in both appveyor and travis, travis failures seem to be gem related.

This PR requires some rebasing prior to merge.

File: Gemfile [hunk 2]
```diff
@@ -17,6 +17,7 @@ ruby_version_segments = Gem::Version.new(RUBY_VERSION.dup).segments
 minor_version = ruby_version_segments[0..1].join('.')
 
 group :development do
+  gem 'serverspec'
```
Comment (tphoney) [inline Gemfile]: not needed

File: Gemfile [hunk 3]
```diff
@@ -28,11 +29,8 @@ group :development do
   gem "puppet-module-win-default-r#{minor_version}", '~> 0.3',   require: false, platforms: [:mswin, :mingw, :x64_mingw]
   gem "puppet-module-win-dev-r#{minor_version}", '~> 0.3',       require: false, platforms: [:mswin, :mingw, :x64_mingw]
   gem "github_changelog_generator",                              require: false, git: 'https://github.com/skywinder/github-changelog-generator', ref: '20ee04ba1234e9e83eb2ffb5056e23d641c7a018' if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.2.2')
-end
-group :system_tests do
-  gem "puppet-module-posix-system-r#{minor_version}", require: false, platforms: [:ruby]
-  gem "puppet-module-win-system-r#{minor_version}",   require: false, platforms: [:mswin, :mingw, :x64_mingw]
-  gem "bolt", '~> 1.3',                               require: false if ENV['GEM_BOLT']
+  # gem "puppet_litmus",                                           require: false, platforms: [:ruby, :mswin, :mingw, :x64_mingw] if ENV['PUPPET_GEM_VERSION'].nil? or ENV['PUPPET_GEM_VERSION'] !~ %r{ 5}
+  gem "puppet_litmus", git: 'https://github.com/michaeltlombardi/puppet_litmus', branch: 'bugfix-localhost-run_bolt_task'
```
Comment (tphoney) [inline Gemfile]: remove before merge (obviously)

File: Gemfile (hunk 2 above)
Comment (tphoney) [inline Gemfile]: not needed, in puppet-module-gems

File: .sync.yml [hunk 4]
```diff
@@ -3,51 +3,27 @@
   required:
```
Comment (tphoney) [inline .sync.yml]: you should run a pdksync as well, to make sure that the next pdksync will not destroy any changes you have put in

**Comment by michaeltlombardi** [comment]
Updated per comments and added a pdksync. I think the appveyor tests will continue to fail until a new version of the litmus gem is released containing the changes from puppetlabs/puppet_litmus#147.

File: provision.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
+default:
+  provisioner: docker_exp
+  images: ['waffleimage/centos7']
+waffle_deb:
+  provisioner: docker_exp
+  images: ['waffleimage/debian8', 'waffleimage/debian9', 'waffleimage/ubuntu14.04', 'waffleimage/ubuntu16.04', 'waffleimage/ubuntu18.04']
+waffle_el6:
```
Comment (tphoney) [inline provision.yaml]: seeing as the tests are quick to run, we should combine the el6 and el7 groups. And just run one group, not 2 for EL

Comment (michaeltlombardi) [reply provision.yaml]: @tphoney Looking at the Travis config, it looks like it is distinguishing between the EL5 and EL6 platforms? If those can be safely squashed I'm happy to do this, digging in now.

**Comment by michaeltlombardi** [comment]
@tphoney yeah I'm gonna add a litmus release to my to-do list for today, at least to prep it all.

**Comment by pmcmaw** [comment]
Resolves #82

**Pull request merged**